src/components/Toast/Toast.jsx

```diff
@@ -0,0 +1,46 @@
+import { Actions } from "../../reducers";
+import { useProduct } from "../../contexts";
+import "./Toast.css";
+
+export const handleToast = (dispatch, text) => {
+  dispatch({
+    type: Actions.OPEN_OR_CLOSE_TOAST,
+    payload: { show: true, text },
+  });
+  setTimeout(() => closeToast(dispatch), 1000);
+};
+
+const closeToast = (dispatch) => {
```

Comment:
Again something minor. Rather than passing `dispatch` as a parameter, you can add these in the Toast component